### PR TITLE
A student cannot become a teacher even on non-PL sections

### PR DESCRIPTION
**Pull request by lfryemason**
## Warning!!

We have entered Pixel Lock for Hour of Code!

Computer Science Education Week will be happening from Dec 4 - Dec 10. Alongside this event, we will
be launching our new Hour of Code activity. Please consider any risk introduced in this PR that
could affect Dance Lab, instructions, saving and logging student progress, caching, or anything
related to the Hour of Code activities, old or new. Even small changes, such as a different button
color, are considered significant during this time. If this change will affect the new Hour of Code
activity in any way, join the morning change review to get your changes approved prior to merging.
Reach out to the Student Labs team for more details!

Fixes two bugs:
* A teacher should not be able to become a coteacher if they are a student even on non-pl sections
* The FE was showing `error unknown` when a student was attempted to be added as a teacher

[Slack thread](https://codedotorg.slack.com/archives/C0462BZKHL6/p1700507153887269?thread_ts=1700493611.607919&cid=C0462BZKHL6)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionsRefresh/coteacherSettings/AddCoteacher.jsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ const getErrorMessageFromResponse = (response, email) => {
       if (json.error.includes('inviting self')) {
         return i18n.coteacherCannotInviteSelf();
       }
-      if (json.error.includes('already in section')) {
+      if (json.error.includes('already a teacher')) {
```
Comment (etaderhold) [inline apps/src/templates/sectionsRefresh/coteacherSettings/AddCoteacher.jsx]:
```suggestion
      if (json.error.includes('already a student')) {
```

**Review by etaderhold — APPROVED** [review]

File: apps/src/templates/sectionsRefresh/coteacherSettings/AddCoteacher.jsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ const getErrorMessageFromResponse = (response, email) => {
       if (json.error.includes('inviting self')) {
         return i18n.coteacherCannotInviteSelf();
       }
-      if (json.error.includes('already in section')) {
+      if (json.error.includes('already a student')) {
         return i18n.coteacherAlreadyInCourse({email});
```
Comment (davidsbailey) [inline apps/src/templates/sectionsRefresh/coteacherSettings/AddCoteacher.jsx:41]: maybe you can fix this in another PR if it's not breaking anything and you're in a hurry, but it looks like the `{email}` param you're passing here is unused

Comment (lfryemason) [reply apps/src/templates/sectionsRefresh/coteacherSettings/AddCoteacher.jsx:41]: Will add as a FLUP

Comment (lfryemason) [reply apps/src/templates/sectionsRefresh/coteacherSettings/AddCoteacher.jsx:41]: Fixed in https://github.com/code-dot-org/code-dot-org/pull/55053/files#diff-4fdbe7e710a3bca0b4c3e608efd9dc965091370a7361e13bc61d9cb6b8b530e6R41

**Pull request merged**